### DISCUSSION TITLE
Changing route paths in sanic or fastapi and set password.

### DISCUSSION THREAD

**Discussion by Hammer2900** [idom-team/idom#537] [Question]
We can now choose fastapi or ssanic as the backend for our interface.
If we write a local application, it works fine, super.
But, if we want to add an interface to a running service and put it into production, we need to specify a path and put a password on this path to protect the application from unauthorized access.
We don't have this option now, but it would be nice to add it.
Do you think it would be possible?

**Comment by rmorshea** [comment]
Authentication isn't something I've gotten around to solving yet. The solution will probably rely on the, [as yet unimplemented](https://github.com/idom-team/idom/issues/203), `use_context` hook to expose the underlying request object within an IDOM component. This might look something like:

```python
from idom import component, html, use_context
from idom.server.sanic import RequestContext

@component
def AuthenticatedComponent():
    request = use_context(RequestContext)
    if not check_auth_somehow(request):
        return html.p("Not authenticated")
    ...
```
